packages/extension-api/src/extension-api.d.ts

```diff
@@ -788,6 +789,29 @@ declare module '@podman-desktop/api' {
     connection: ContainerProviderConnection;
   }
 
+  export interface ProviderConnectionShellAccess {
+    onData: Event<ProviderConnectionShellAccessData>;
+    onError: Event<ProviderConnectionShellAccessError>;
+    onEnd: Event<void>;
+    write(data: string): void;
+    startConnection(): void;
+    stopConnection(): void;
+    setWindow(dimensions: ShellDimensions): void;
+  }
+
+  export interface ShellDimensions {
```

Comment:
should be prefixed for now as well
```suggestion
  export interface ProviderConnectionShellDimensions {
```